diff --git a/bark.cpp b/bark.cpp
index 6e99d99..afaf071 100644
--- a/bark.cpp
+++ b/bark.cpp
@@ -23,6 +23,9 @@
 #include <vector>
 
 #include "bark.h"
+
+#include <ggml-cpu.h>
+
 #include "encodec.h"
 
 #define BARK_MAX_NODES 4096
@@ -303,9 +306,9 @@ static bool ggml_quantize_weights(
         case GGML_FTYPE_MOSTLY_IQ4_XS:
         case GGML_FTYPE_MOSTLY_IQ1_M:
         case GGML_FTYPE_MOSTLY_BF16:
-        case GGML_FTYPE_MOSTLY_Q4_0_4_4:
-        case GGML_FTYPE_MOSTLY_Q4_0_4_8:
-        case GGML_FTYPE_MOSTLY_Q4_0_8_8:
+        // case GGML_FTYPE_MOSTLY_Q4_0_4_4:
+        // case GGML_FTYPE_MOSTLY_Q4_0_4_8:
+        // case GGML_FTYPE_MOSTLY_Q4_0_8_8:
                 {
                     fprintf(stderr, "%s: invalid model type %d\n", __func__, ftype);
                     return false;
@@ -446,9 +449,9 @@ static bool ggml_quantize_weights(
                 case GGML_TYPE_IQ4_XS:
                 case GGML_TYPE_IQ1_M:
                 case GGML_TYPE_BF16:
-                case GGML_TYPE_Q4_0_4_4:
-                case GGML_TYPE_Q4_0_4_8:
-                case GGML_TYPE_Q4_0_8_8:
+                // case GGML_TYPE_Q4_0_4_4:
+                // case GGML_TYPE_Q4_0_4_8:
+                // case GGML_TYPE_Q4_0_8_8:
                 case GGML_TYPE_TQ1_0:
                 case GGML_TYPE_TQ2_0:
                 case GGML_TYPE_COUNT:
