diff --git a/ggml_extend.hpp b/ggml_extend.hpp
index c5913be..02c0af7 100644
--- a/ggml_extend.hpp
+++ b/ggml_extend.hpp
@@ -113,7 +113,8 @@ __STATIC_INLINE__ struct ggml_tensor* ggml_kronecker(ggml_context* ctx, struct g
                                      a->ne[0] * b->ne[0],
                                      a->ne[1] * b->ne[1],
                                      a->ne[2] * b->ne[2],
-                                     a->ne[3] * b->ne[3]),
+                                     a->ne[3] * b->ne[3],
+                                     ggml_scale_mode::GGML_SCALE_MODE_NEAREST),
                     b);
 }
 
